src/frontend/src/utils/condition.rs

```diff
@@ -382,11 +391,108 @@ impl Condition {
+                    if let Some(f_input) = f.inputs().get(0)
+                        && let Some(f_input) = f_input.as_function_call()
+                        && matches!(f_input.func_type(), ExprType::Row)
+                    {
+                        true
+                    } else {
+                        false
+                    }
+                } else {
+                    false
+                }
+            });
+
+        // optimize for single row conjunctions. More optimisations may come later
+        // For example, (v1,v2,v3) > (1, 2, 3) means all data from (1, 2, 3).
+        // Suppose v1 v2 v3 are both pk, we can push (v1,v2,v3）> (1,2,3) down to scan
+        // Suppose v1 v2 are both pk, we can push (v1,v2）> (1,2) down to scan and add (v1,v2,v3) > (1,2,3) in filter, it is still possible to reduce the value of scan
+        if row_conjunctions.len() == 1 {
+            let row_conjunction = row_conjunctions.pop().unwrap();
+            let row_left_inputs = row_conjunction
+                .as_function_call()
+                .unwrap()
+                .inputs()
+                .get(0)
+                .unwrap()
+                .as_function_call()
+                .unwrap()
+                .inputs();
+            let row_right_literal = row_conjunction
+                .as_function_call()
+                .unwrap()
+                .inputs()
+                .get(1)
+                .unwrap()
+                .as_literal()
+                .unwrap();
+            let row_right_literal_data = row_right_literal.get_data().clone().unwrap();
+            let row_right_literal_type = row_right_literal.get_data_type().clone().unwrap();
+            let right_iter = row_right_literal_data
+                .as_struct()
+                .fields()
+                .iter()
+                .zip_eq_fast(row_right_literal_type.as_struct().types());
+            let func_type = row_conjunction.as_function_call().unwrap().func_type();
+            if row_left_inputs.len() > 1
+                && (matches!(func_type, ExprType::LessThan)
+                    || matches!(func_type, ExprType::GreaterThan))
+            {
+                let mut row_inner_element_map = row_left_inputs
+                    .iter()
+                    .zip_eq_fast(right_iter)
+                    .map(|(left_expr, right_expr)| {
+                        (
+                            left_expr.as_input_ref().unwrap().index,
+                            (left_expr, right_expr),
+                        )
+                    })
+                    .collect::<HashMap<_, _>>();
+                let mut pk_struct = vec![];
+                for i in table_desc.order_column_indices() {
+                    if let Some((_, (data, _))) = row_inner_element_map.remove(&i) {
+                        pk_struct.push(data.clone());
+                    } else {
+                        break;
+                    }
+                }
```

Comment:
Considering the primary key is (a, b, c), if users write a predicate like `(c, b, a) > (1, 2, 3)`, it seems we will treat it as `(a, b, c) > (3, 2, 1)` here. However, it is incorrect.